covquestions-js/models/Questionnaire.generated.ts

```diff
@@ -464,3 +468,45 @@ export interface Variable {
   id: string;
   expression: LogicExpression;
 }
+/**
+ * One test case that simulates a virtual patient and the results they should receive.
+ */
+export interface TestCase {
+  /**
+   * The description of the test case.
+   */
+  description: string;
+  /**
+   * Object of given answers in the form: 'questionId: answer'. Answer can be the answerId for selects, a value, or an array for multi-selects.
+   */
+  answers: {
+    [k: string]: string | number | boolean | any[];
+  };
+  /**
+   * Object of obtained results in the form: 'resultCategoryId: resultId'.
+   */
+  results: {
+    [k: string]: string;
+  };
+  /**
+   * Options to configure the behavior of the test case runner.
+   */
+  options?: {
+    /**
+     * The simulated time of execution. Important for date questions, where the evaluation looks for time periods, like the last 14 days.
+     */
+    fillInDate?: string;
+    /**
+     * If not set, ignores answers that are provided in the test case, but were not needed in the questionnaire. It also tries to 'skip' optional questions if no answer is provided. In strict mode, each appearing question has to be set in the test case and each answer in the test case has to be given in the questionnaire. In random mode, answers that are not provided are randomly chosen, answers that were provided, but not used are ignored.
+     */
+    questionMode?: "strict" | "random";
```

Comment:
Does that mean that by default, optional questions are never answered?

Reply:
Okay, read the code and and now it's clear.

Maybe we can change the doc to something like "If not set, answers for optional questions might be omitted. If set to 'strict', answers for all not skipped questions need to be provided. ...".

Reply:
I changed the implementation for now to not require optional questions to be set in strict mode. I changed the doc accordingly. Maybe we can remove the "normal" mode once the random mode is ready. Let's see...